app/src/lib/stores/app-store.ts

```diff
@@ -1606,13 +1607,63 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
       return { workingDirectory, selectedFileIDs, diff }
     })
+
+    this._triggerMergeConflictsFlow(repository, status)
+
     this.emitUpdate()
 
     this.updateChangesDiffForCurrentSelection(repository)
 
     return true
   }
 
+  /** starts the conflict resolution flow, if appropriate */
+  private async _triggerMergeConflictsFlow(
```

Comment:
One thing I've forgotten to ask about - should this flow be feature-flagged behind `enableMergeConflictsDialog()`? The old version of the app didn't show any UI in this situation, and I think we should do the same here.

Reply:
that's a good question. this will run but the popup won't show because the feature flag is checked when the app is rendering, so i think we're good (https://github.com/desktop/desktop/blob/4c0aaa7e4b36e1ca34c5a17da0df3c8c71a177e7/app/src/ui/app.tsx#L1334)

Reply:
But a dialog is still shown on the `else` path:

https://github.com/desktop/desktop/blob/15e4f99570eb90a20c3689f4c532841e3a6b2d7e/app/src/ui/app.tsx#L1367-L1375

Reply:
59e4773